PrestaShop/PrestaShop
# Make migrated order page's grid rows clickable

| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Make row clickable in migrated order page. Also make which column are clickable inside a clickable row configurable. 
| Type?         | Improvement
| Category?     | BO
| BC breaks?    | yes, se below
| Deprecations? | no
| Fixed ticket? | Fixes #14736
| How to test?  | See below

#### Feature description

Grid rows for Orders listing ( `/admin-dev/sell/orders/orders/` ) should be clickable. However the columns which have already some actions (links, dropdown, buttons ...) do not have the clickable behavior. So only columns without action are clickable.

#### PR Scope

We actually rewrote the "clickable row" behavior used in **all grids** to use the same everywhere.

#### How to test ?

Check that rows are clickable in migrated order pages `/admin-dev/sell/orders/orders/` except for columns. Check that other grids have a similar behavior.

#### BC Break ?

Before, the "clickable row" behavior was available on column/rows with selectors `td.data-type, td.identifier-type:not(:has(input)), td.badge-type, td.position-type` . We removed these to replace them with the `td.clickable` selector and made sure that standard PrestaShop columns have this class.

However if a developer used the Grid component to build its own column, this might be a BC break for him when he upgrades PS as the "clickable" behavior will only work if his column get the new selector too.




---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/prestashop/prestashop/16318)
